### DISCUSSION TITLE
Alternative API for io_uring_register_buffers for multiple rings with shared backend

### DISCUSSION THREAD

**Discussion by yash760** [axboe/liburing#892] [Q&A] [answered]
I have a HugePage memory backed application that can receive IO on 30 threads. All these 30 threads have their own ring which have shared backend worker thread (**io_uring_shared_wq**). Given these conditions I want to register my complete HugePage memory on initialisation with the rings of all the threads using **io_uring_register_buffers** .
Is there any other alternative API other than to call **io_uring_register_buffers** for each ring of 30 threads. Also does calling register buffer only on the shared worker queue ring works ?

**Comment by axboe** [comment, marked as answer]
Registered buffers can only be used on a ring where they have been registered. So you'd need to register the same buffer on all rings where you'd be doing IO to/from registered buffers with read/write fixed.

**Comment by yash760** [comment]
Thanks for the answer @axboe !!! Had a follow up question on this.
How much memory will be used by the library/ring for storing the the `struct iovec` that point to the registered buffers. For example lets say I register 1024 such iovec, then how much memory utilization will increase? I tried checking the size of ring by using `io_uring_mlock_size` before and after registering buffers and it returned the same value.

**Comment by axboe** [comment]
The library doesn't store anything, the kernel side overhead is 16 bytes per registered buffer. So 1024 buffers => 16kb of memory.
